app/Libraries/BeatmapsetDiscussionPostHandlesProblem.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries;
+
+use App\Jobs\Notifications;
+use App\Models\BeatmapDiscussion;
+use App\Models\User;
+
+abstract class BeatmapsetDiscussionPostHandlesProblem
+{
+    protected int $priorOpenProblemCount = 0;
+    protected ?BeatmapDiscussion $problemDiscussion = null;
```

Comment:
this feels like should be called `newProblemDiscussion` as the other problem discussion handling (resolved/reopen) isn't using this

...and maybe rename the class to `...HandlesNewProblem`? And remove the `wasRecentlyCreated`/`exists` check

there may be further simplification possible (like moving the prior open problems check to right before assigning new problem discussion and removing the count variable entirely - or changing it as a cache variable)

```php
// both in review and non-review assignment
// app/Libraries/BeatmapsetDiscussionReview.php#148
// app/Libraries/BeatmapsetDiscussionPostNew.php#51
$this->shouldRecordNewProblem() { // or maybe roll this and next line into single function
  $this->recordNewProblem($discussion);
}

function shouldRecordNewProblem()
{
  return $this->hasPriorOpenProblems ??= ...openProblems()->exists();
}

function recordNewProblem($discussion)
{
    if ($this->newProblemDiscussion === null && ($discussion->message_type !== 'problem' || !$this->shouldRecordNewProblem())) {
        return;
    }
    $this->newProblemDiscussion = $discussion;
}
```

Reply:
I don't see how this helps; disqualification and nomination reset depend on whether the discussion is new and the user has the right permissions, regardless of whether there are existing problems; the problem notification depends on whether there are existing problems regardless of it being a reply or new discussion but only if the action didn't result in it being disqualified or reset.

Reply:
oh huh. Also apparently I missed the confirmation thing for reopening issue ಠ_ಠ